src/w_wad.c

```diff
@@ -242,7 +242,7 @@ static void W_CoalesceMarkedResource(const char *start_marker,
     {
       lumpinfo[numlumps].size = 0;  // killough 3/20/98: force size to be 0
       lumpinfo[numlumps].namespace = ns_global;   // killough 4/17/98
-      memcpy(lumpinfo[numlumps++].name, end_marker, 8);
+      M_StringCopy(lumpinfo[numlumps++].name, end_marker, 8);
```

Comment:
This is effectively `strncpy(lumpinfo[numlumps++].name, end_marker, 7)`, i.e. copying only 7 chars of the lump name.

Reply:
What was the actual asan report here?

Reply:
Asan report:
<details>

```
=================================================================
==3916==ERROR: AddressSanitizer: global-buffer-overflow on address 0x7fff7138f2c6 at pc 0x7fff6e744e5b bp 0x009ce07cf740 sp 0x009ce07ceec0
READ of size 8 at 0x7fff7138f2c6 thread T0
    #0 0x7fff6e744e88  (C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.35.32215\bin\HostX64\x64\clang_rt.asan_dbg_dynamic-x86_64.dll+0x180044e88)
    #1 0x7fff7107cc35  (c:\develop\woof\build_asan\src\woof.dll+0x18018cc35)
    #2 0x7fff71079557  (c:\develop\woof\build_asan\src\woof.dll+0x180189557)
    #3 0x7fff70f14883  (c:\develop\woof\build_asan\src\woof.dll+0x180024883)
    #4 0x7fff70f403d6  (c:\develop\woof\build_asan\src\woof.dll+0x1800503d6)
    #5 0x7ff6bfa414db  (c:\develop\woof\build_asan\src\woof.com+0x1400014db)
    #6 0x7ff6bfa4532b  (c:\develop\woof\build_asan\src\woof.com+0x14000532b)
    #7 0x7ff6bfa453d2  (c:\develop\woof\build_asan\src\woof.com+0x1400053d2)
    #8 0x7ff6bfa41958  (c:\develop\woof\build_asan\src\woof.com+0x140001958)
    #9 0x7ff6bfa417fd  (c:\develop\woof\build_asan\src\woof.com+0x1400017fd)
    #10 0x7ff6bfa416bd  (c:\develop\woof\build_asan\src\woof.com+0x1400016bd)
    #11 0x7ff6bfa419ed  (c:\develop\woof\build_asan\src\woof.com+0x1400019ed)
    #12 0x7fff9c4b7603  (C:\Windows\System32\KERNEL32.DLL+0x180017603)
    #13 0x7fff9d5226a0  (C:\Windows\SYSTEM32\ntdll.dll+0x1800526a0)

0x7fff7138f2c6 is located 0 bytes to the right of global variable '<C++ string literal>' defined in 'w_wad.c:385:38' (0x7fff7138f2c0) of size 6
  '<C++ string literal>' is ascii string 'S_END'
SUMMARY: AddressSanitizer: global-buffer-overflow (C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.35.32215\bin\HostX64\x64\clang_rt.asan_dbg_dynamic-x86_64.dll+0x180044e88)
Shadow bytes around the buggy address:
  0x11f73e171e00: f9 f9 f9 f9 f9 f9 f9 f9 05 f9 f9 f9 f9 f9 f9 f9
  0x11f73e171e10: f9 f9 f9 f9 f9 f9 f9 f9 00 00 00 00 00 02 f9 f9
  0x11f73e171e20: f9 f9 f9 f9 f9 f9 f9 f9 00 00 00 00 06 f9 f9 f9
  0x11f73e171e30: f9 f9 f9 f9 f9 f9 f9 f9 00 00 00 00 01 f9 f9 f9
  0x11f73e171e40: f9 f9 f9 f9 f9 f9 f9 f9 00 00 00 04 f9 f9 f9 f9
=>0x11f73e171e50: f9 f9 f9 f9 f9 f9 f9 f9[06]f9 f9 f9 f9 f9 f9 f9
  0x11f73e171e60: f9 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9
  0x11f73e171e70: f9 f9 f9 f9 f9 f9 f9 f9 06 f9 f9 f9 f9 f9 f9 f9
  0x11f73e171e80: f9 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9
  0x11f73e171e90: f9 f9 f9 f9 f9 f9 f9 f9 06 f9 f9 f9 f9 f9 f9 f9
  0x11f73e171ea0: f9 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
  Shadow gap:              cc
==3916==ABORTING
```
</details>

Reply:
Done https://github.com/fabiangreffrath/woof/pull/1012/commits/48e54375cb8ab350126e4a6f9dd908022e641066